Source/UZKFileInfo.m

```diff
@@ -105,4 +107,20 @@ - (NSDate *)readDate:(tm_unz)date
     return [calendar dateFromComponents:components];
 }
 
+- (BOOL)isDirectoryWith:(unz_file_info64 *)fileInfo {
+    
+    uLong type = fileInfo->external_fa >> 0x1D & 0x1F;
```

Comment:
suggest judging from the implementation of the new code from the link I mentioned above, which can improve the correctness of the judgment. The code I am currently submit still needs a test example to verify. But since I am still on vacation, I am temporarily unable to submit new code, I think you can help me.